### PR TITLE
[sgl-kernel/cpu] support w8a8 int8 model for arm cpu

### PR DESCRIPTION
**Pull request by cyb70289**
## Motivation

Support W8A8 Int8 Dense and MoE model for Arm CPU.

## Accuracy Tests

Tested accuracy with few_shot_gsm8k dataset, 200 questions.

```
Dense model
-----------
BF16: meta-llama/Llama-3.2-3B-Instruct
Accuray: 0.680

W8A8: RedHatAI/Llama-3.2-3B-Instruct-quantized.w8a8
Accuray: 0.670

MoE model
---------
BF16: Qwen/Qwen3-30B-A3B-Instruct-2507
Accuray: 0.945

W8A8: ramblingpolymath/Qwen3-30B-A3B-Instruct-2507-W8A8
Accuray: 0.955
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

### CODE REVIEW

**Comment by cyb70289** [comment]
depends on https://github.com/sgl-project/sglang/pull/14867

File: python/sglang/srt/layers/quantization/w8a8_int8.py [hunk 1]
```diff
@@ -338,10 +339,11 @@ def apply(
         x = dispatch_output.hidden_states
         topk_output = dispatch_output.topk_output
 
-        if use_intel_amx_backend(layer):
+        if use_intel_amx_backend(layer) or _is_cpu_arm64:
             from sglang.srt.layers.moe.topk import apply_topk_weights_cpu
 
             topk_weights, topk_ids, _ = topk_output
+            topk_ids = topk_ids.int()
```
Comment (cyb70289) [inline python/sglang/srt/layers/quantization/w8a8_int8.py:361]: this is also necessary for original amx code

File: sgl-kernel/csrc/cpu/CMakeLists.txt [hunk 2]
```diff
@@ -27,17 +27,32 @@ include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}
 )
 
-# Platform-specific library directory
+# Build all cpp files in current and sub dirs
```
Comment (cyb70289) [inline sgl-kernel/csrc/cpu/CMakeLists.txt:30]: Original code includes all *.cpp files under current and subdirs.
This PR supprts arch dependent subdirs (e.g., aarch64, x86_64, ppc64), and only include *.cpp under matched arch subdir to project.

File: python/sglang/srt/layers/quantization/w8a8_int8.py [hunk 3]
```diff
@@ -310,10 +314,7 @@ def create_weights(
         layer.register_parameter("w2_input_scale", w2_input_scale)
 
     def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
-        if _is_cpu:
```
Comment (alexnails) [inline python/sglang/srt/layers/quantization/w8a8_int8.py:313]: this breaks other cpu using is_cpu ?

**Comment by mingfeima** [comment]
rebase and run ci again

File: sgl-kernel/csrc/cpu/gemm_int8.cpp [hunk 4]
```diff
@@ -473,6 +473,7 @@ at::Tensor int8_scaled_mm_cpu(
   return out;
 }
 
+#ifndef __aarch64__
```
Comment (alexnails) [inline sgl-kernel/csrc/cpu/gemm_int8.cpp:476]: why was this previously not restricted to arch64? are we causing regression here?

**Review by alexnails — COMMENTED** [review]
forgot to publish my comments apologies

File: sgl-kernel/csrc/cpu/gemm_int8.cpp (hunk 4 above)
Comment (cyb70289) [reply sgl-kernel/csrc/cpu/gemm_int8.cpp:476]: Hmm..., I shall leave this file untouched. Just skip building it on arm platform. Will propose a PR to improve. Thanks for pointing out.

File: python/sglang/srt/layers/quantization/w8a8_int8.py (hunk 3 above)
Comment (cyb70289) [reply python/sglang/srt/layers/quantization/w8a8_int8.py:313]: Original code asserts if runs on cpu but not amx. Looks it does no harm just let non-amx cpu go through the default no-prepacking path?
